packages/apps/plugins/plugin-debug/src/components/DebugSpace.tsx

```diff
@@ -46,18 +53,21 @@ const DebugSpace: FC<{ space: Space; onAddObjects?: (objects: ReactiveObject<any
   const { connect } = useSpaceInvitation(space?.key);
   const client = useClient();
   const [data, setData] = useState<any>({});
-  const handleRefresh = async () => {
-    const data = await client.diagnostics({ truncate: true });
-    setData(
-      data?.diagnostics?.spaces?.find(({ key }: any) => {
-        return space.key.toHex().startsWith(key);
-      }),
-    );
-  };
 
-  useEffect(() => {
-    void handleRefresh();
-  }, [space]);
+  const [update, handleUpdate] = useRefresh();
+  useAsyncEffect(
```

Comment:
we've been avoiding this for a while, I'm surprised it's not marked as deprecated

Reply:
generally we've been using

```
useEffect(() => {
  const timeout = setTimeout(async () => {
    // your async code here
  });

  return () => clearTimeout(timeout);
}, []);
```

Reply:
how do you check if it's mounted?

Reply:
the timeout is cleared if its unmounted so it isn't run on unmounted component, is that the problem you're talking about?